golang.sh

```diff
@@ -0,0 +1,47 @@
+package: golang
+version: "1.10.2"
+build_requires:
+  - curl
+prefer_system_check: |
+  which go && case `go version | sed -e 's/go version go//' | sed -e 's/ .*//'` in 0*|1.[0-9].*) exit 1 ;; esac
```

Comment:
Do not use `which`, call `go version` directly instead (it returns `0` in case `go` exists). I know, this discussion was done already in other places: `which` might _(might)_ not be available on all platforms and in all cases it's better to invoke the target executable directly.